include/utility/json_utility.hpp

```diff
@@ -650,5 +650,141 @@ inline bool isBackupAndRestoreRequired(const nlohmann::json& i_sysCfgJsonObj)
+ * and "value" tags under the given VPD file path. And std::runtime_error for
+ * invalid parameters.
+ *
+ * @param[in] i_sysCfgJsonObj -  System config JSON object.
+ * @param[in] i_vpdFilePath - VPD file path.
+ *
+ * @return GPIO parameters if present, otherwise throws std::runtime_error.
+ */
+inline types::GpioPollingParameters
+    getGpioPollingParameters(const nlohmann::json& i_sysCfgJsonObj,
+                             const std::string i_vpdFilePath)
+{
+    types::GpioPollingParameters l_pollingParameters;
+
+    if (i_sysCfgJsonObj.empty() || i_vpdFilePath.empty())
+    {
+        throw std::runtime_error("Invalid parameters");
+    }
+
+    if (!i_sysCfgJsonObj.contains("frus"))
+    {
+        throw std::runtime_error("Missing frus section in system config JSON");
+    }
+
+    if (i_sysCfgJsonObj["frus"][i_vpdFilePath]
+            .at(0)["pollingRequired"]
+            .contains("gpioPresence"))
+    {
+        std::get<0>(l_pollingParameters) =
+            i_sysCfgJsonObj["frus"][i_vpdFilePath].at(
+                0)["pollingRequired"]["gpioPresence"]["pin"];
+        std::get<1>(l_pollingParameters) =
+            i_sysCfgJsonObj["frus"][i_vpdFilePath].at(
+                0)["pollingRequired"]["gpioPresence"]["value"];
+        std::get<2>(l_pollingParameters) =
+            i_sysCfgJsonObj["frus"][i_vpdFilePath].at(0)["inventoryPath"];
+    }
+    else
+    {
+        throw std::runtime_error(
+            "Missing gpioPresence tag under pollingRequired");
+    }
+
+    return l_pollingParameters;
+}
+
+/**
+ * @brief An API returns list of FRUs parameters that needs Polling.
+ *
+ * An API that checks for the FRUs that requires GPIO polling and returns
+ * a list of FRUs with polling parameters. Returns an empty list if there are
+ * no FRUs that requires polling.
+ *
+ * @param[in] i_sysCfgJsonObj - System config JSON object.
+ *
+ * @return list of FRUs parameters that needs polling.
+ */
+inline std::vector<types::GpioPollingParameters>
+    getPollingRequiredFrusDetails(const nlohmann::json& i_sysCfgJsonObj)
+{
+    std::vector<types::GpioPollingParameters> l_pollingRequiredFrusParamsList;
+
+    if (i_sysCfgJsonObj.empty())
+    {
+        throw std::runtime_error("Invalid Parameters");
+    }
+
+    if (!i_sysCfgJsonObj.contains("frus"))
+    {
+        throw std::runtime_error("Missing frus section in system config JSON");
+    }
+
+    for (const auto& l_fru : i_sysCfgJsonObj["frus"].items())
+    {
+        const auto l_fruPath = l_fru.key();
+
+        if (jsonUtility::isPollingRequired(i_sysCfgJsonObj, l_fruPath))
+        {
+            try
+            {
+                types::GpioPollingParameters l_pollingParameters =
+                    jsonUtility::getGpioPollingParameters(i_sysCfgJsonObj,
+                                                          l_fruPath);
+                l_pollingRequiredFrusParamsList.push_back(l_pollingParameters);
```

Comment:
Why are we pushing empty tuples into the list?

Reply:
Throws error if "pollingRequired" tag is missing